### PR TITLE
[12][ADD] new module hr_employee_partner_external

### PR DESCRIPTION
**Pull request by mymage**
I propose this module to have the possibility to add a reference to a partner for the employee.
This can be useful if you have to manage people who are in charge of a partner or if you want to assign some activity to a "generic" person who is managed by the partner.

### CODE REVIEW

**Review by marcelofrare — APPROVED** [review]
I've test it. Working done ok!

**Review by andreampiovesana — APPROVED** [review]
go

File: hr_employee_partner/models/hr_employee.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2020 Stefano Consolaro (Ass. PNLUG - Gruppo Odoo <http://odoo.pnlug.it>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class EployeePartner(models.Model):
```
Comment (etobella) [inline hr_employee_partner/models/hr_employee.py]: Typo error

File: hr_employee_partner/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "license":  "AGPL-3",
+
+    "depends": [
+        'hr',
+        ],
+    "data": [
+        'views/hr_employee.xml',
+        ],
+    'installable': True,
+    'auto_install': False,
```
Comment (etobella) [inline hr_employee_partner/__manifest__.py]: Not required auto_install and installable

File: hr_employee_partner/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Create an Employee as real person or a "virtual" one and set the Partner Complany.
```
Comment (etobella) [inline hr_employee_partner/readme/USAGE.rst]: Typo error

File: hr_employee_partner/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+This module adds the following to employees :
```
Comment (etobella) [inline hr_employee_partner/readme/DESCRIPTION.rst]: Add the following fields to employee

**Review by etobella — COMMENTED** [review]
Just a few typos.

Just a question, should this partner be the partner of the related user if defined? I mean, the employee has an address home, so I assume this partner isn't personal, so it should be the internal information, isn't it?

File: hr_employee_partner/readme/DESCRIPTION.rst (hunk 4 above)
Comment (mymage) [reply hr_employee_partner/readme/DESCRIPTION.rst]: Ok

File: hr_employee_partner/readme/USAGE.rst (hunk 3 above)
Comment (mymage) [reply hr_employee_partner/readme/USAGE.rst]: Thanks

File: hr_employee_partner/__manifest__.py (hunk 2 above)
Comment (mymage) [reply hr_employee_partner/__manifest__.py]: Many things of Odoo are still on the dark side for me :-). Thanks

File: hr_employee_partner/models/hr_employee.py (hunk 1 above)
Comment (mymage) [reply hr_employee_partner/models/hr_employee.py]: Thanks

**Comment by mymage** [comment]
> Just a question, should this partner be the partner of the related user if defined? I mean, the employee has an address home, so I assume this partner isn't personal, so it should be the internal information, isn't it?

@etobella As Employee I identify everyone works inside (o for) the Company, and are managed by the Company management.
There can be some people that a Partner supplies to the Company for cover some jobs.
Company will see the Partner-Employee as a Company-Employee, except for the administration aspects.
So I need all the features of a Company-employee (eg. Skills) and I wanto to know that is administrated by the Partner.
Do you think that it is better to set up a Partner-Contact and an Company-Employee and then join them, than assign the Partner in the Company-Employee?

**Comment by mymage** [comment]
@etobella I discovered that in Employee ther is a field address_id (Work address) that point to Partner (label ?!?!?!?). Now I will investigate what is used for: I imagine that is the location wher a "my employee works"; if so it is different from my intent.

**Comment by jarroyomorales** [comment]
@mymage Hi! If im not understanding it wrong, this field would be to manage employees that work in the company but dont belong to it. In that case it's different from `address_id` and i would change the name of the module to `hr_employee_external_partner`or something that defines better the usage of the module.

**Comment by mymage** [comment]
@etobella @jarroyomorales I verified that address_id isn't what I mean. It is better to havve an explicit reference to the fact that the Employee is administrated by a Partner, but is managed by the company.

[OUTDATED Instead to assign a Partner, we can assign a Partner's contact to the Employee but for me this increase complexity (employee->contact->partner) and work (adding every Partner's Employee as a contact). Maybe this can be another module, useful for assign web portal features to a Partner's Employee. ]
I didn't remember that in Odoo Partners and Partners Contacts are listed together within partner_id (I don't like this), so this proposal is already embedded. It is only a matter of data management.

What do you think?

PS: changed module and field name

**Comment by mymage** [comment]
I added an additional field "Is an external Employee" (is_external) to better emphasize that Employee is not of the Company. This field drive the display of Partner field.

**Review by marcelofrare — APPROVED** [review]
Ok, excelent choice. Work good.

**Review by andreampiovesana — APPROVED** [review]
ok, useful to manage external people

File: hr_employee_partner_external/views/hr_employee.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    <record id="employee_form_inherit_partner" model="ir.ui.view">
+        <field name="name">hr.employee</field>
+        <field name="model">hr.employee</field>
+        <field name="inherit_id" ref="hr.view_employee_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='department_id']" position="after">
+                <field name="is_external"/>
+                <field name="hr_external_partner_id"
+                    attrs="{'invisible':[('is_external','=',False)],
+                            'required': [('is_external','=', True)]}"/>
```
Comment (etobella) [inline hr_employee_partner_external/views/hr_employee.xml:17]: it is not needed to be required, I mean, he could be a freelance, so, the external partner is himself.

What do you think?

**Review by etobella — COMMENTED** [review]
LGTM, but I think the required is not necessary

File: hr_employee_partner_external/views/hr_employee.xml (hunk 5 above)
Comment (mymage) [reply hr_employee_partner_external/views/hr_employee.xml:17]: If he / she is a freelancer, he must be registered as a Partner to fill in the related tax / payment details, so I think it is better in any case to connect the partner's reference.
You register him as an Employee to manage "functional data / procedures" without worrying about administrative things, which will be managed through the Partner link.

**Review by etobella — APPROVED** [review]

**Comment by andreampiovesana** [comment]
merge?

**Review by jarroyomorales — APPROVED** [review]

File: hr_employee_partner_external/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Stefano Consolaro (Ass. PNLUG - Gruppo Odoo <http://odoo.pnlug.it>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name":     "Employee external Partner",
+    "summary":  "Associate an external Partner to Employee",
+    "version":  "12.0.1.0.0",
+
```
Comment (Saran440) [inline hr_employee_partner_external/__manifest__.py]: please, delete line in this file
```suggestion
```

**Review by Saran440 — APPROVED** [review]
Functional Test. LGTM :+1:

File: hr_employee_partner_external/__manifest__.py (hunk 6 above)
Comment (mymage) [reply hr_employee_partner_external/__manifest__.py]: Done, thanks.

**Pull request merged**